### ISSUE TITLE
ShipToAddress for Model CustomerRetDto, must be list

### ISSUE THREAD

**Issue by rashidabbas786** [sflagp/QBXML-Object-Models#7]
Hello,

I noticed one more issue which is related to ShipToAddress inside CustomerRetDto Model. The ShipToAddress must be a list instead of object.
Official Docs says there could be around 50 addresses. 
https://developer.intuit.com/app/developer/qbdesktop/docs/api-reference/qbdesktop/customerquery

Would that be possible for you to directly publish the change? Or it takes time?

**Comment by sflagp** [comment]
I'm sure you'll find more many things like that. I'm sure I used objects instead of list of objects and the other way around too. Thanks for finding that. Let me know when you find more.

[QbModels.1.2.9-beta-01.zip](https://github.com/sflagp/QBXML-Object-Models/files/14088825/QbModels.1.2.9-beta-01.zip)

**Comment by rashidabbas786** [comment]
Please also consider a propery for adress object. **DefaultShipTo**

 ```
<ShipToAddress> <!-- must occur 0 - 50 times -->
                                        <Name >STRTYPE</Name> <!-- required -->
                                        <Addr1 >STRTYPE</Addr1> <!-- optional -->
                                        <Addr2 >STRTYPE</Addr2> <!-- optional -->
                                        <Addr3 >STRTYPE</Addr3> <!-- optional -->
                                        <Addr4 >STRTYPE</Addr4> <!-- optional -->
                                        <Addr5 >STRTYPE</Addr5> <!-- optional -->
                                        <City >STRTYPE</City> <!-- optional -->
                                        <State >STRTYPE</State> <!-- optional -->
                                        <PostalCode >STRTYPE</PostalCode> <!-- optional -->
                                        <Country >STRTYPE</Country> <!-- optional -->
                                        <Note >STRTYPE</Note> <!-- optional -->
                                        <DefaultShipTo >BOOLTYPE</DefaultShipTo> <!-- optional -->
                                </ShipToAddress>
```

**Comment by rashidabbas786** [comment]
Wow that was speedy. 

Have you handled DefaultShipTo? in the List

**Comment by sflagp** [comment]
I happened to be looking at my computer when the notification came in so it was a quick fix. The funny thing is that the Add/Mod requests were already List<AddressDto> with properties to limit it to 50 entries.

I do not see DefaultShipTo in the models anywhere. Where is the documentation on that?

**Comment by rashidabbas786** [comment]
If you go to xml response here:
https://developer.intuit.com/app/developer/qbdesktop/docs/api-reference/qbdesktop/customerquery

You will find the following chunk of xml, check the last property of that object:

```
<ShipToAddress> <!-- must occur 0 - 50 times -->
                                       <Name >STRTYPE</Name> <!-- required -->
                                       <Addr1 >STRTYPE</Addr1> <!-- optional -->
                                       <Addr2 >STRTYPE</Addr2> <!-- optional -->
                                       <Addr3 >STRTYPE</Addr3> <!-- optional -->
                                       <Addr4 >STRTYPE</Addr4> <!-- optional -->
                                       <Addr5 >STRTYPE</Addr5> <!-- optional -->
                                       <City >STRTYPE</City> <!-- optional -->
                                       <State >STRTYPE</State> <!-- optional -->
                                       <PostalCode >STRTYPE</PostalCode> <!-- optional -->
                                       <Country >STRTYPE</Country> <!-- optional -->
                                       <Note >STRTYPE</Note> <!-- optional -->
                                       <DefaultShipTo >BOOLTYPE</DefaultShipTo> <!-- optional -->
                               </ShipToAddress>```

**Comment by sflagp** [comment]
I see it now. That will require a little more work but I can take care of that. I will let you know when I have an update.

**Comment by rashidabbas786** [comment]
Sure that Would be awesome.

**Comment by rashidabbas786** [comment]
How do you publish over nuget package manager? how long it takes to publically available

**Comment by sflagp** [comment]
I just uploaded the last beta version to nuget. It depends on them when it is available but it usually takes about an hour to be accepted. I will let you know when I get the email that it's published.

**Comment by rashidabbas786** [comment]
Perfect. Thank you so much <3

**Comment by sflagp** [comment]
That was quick. I got the email that it was published and verified that I can see it in nuget as a prerelease version.

**Comment by rashidabbas786** [comment]
Wow. Thank you so much, brother.  Updating my package.

**Comment by sflagp** [comment]
v1.2.9 is now on nuget which should have the DefaultShipTo for Customer. Let me know if you have any issues.

**Comment by rashidabbas786** [comment]
Yes. Its there now. Thank you :-)

**Comment by rashidabbas786** [comment]
> If you go to xml response here: https://developer.intuit.com/app/developer/qbdesktop/docs/api-reference/qbdesktop/customerquery
> 
> You will find the following chunk of xml, check the last property of that object:
> 
> ```
> <ShipToAddress> 
>                                        <Name >STRTYPE</Name> 
>                                        <Addr1 >STRTYPE</Addr1> 
>                                        <Addr2 >STRTYPE</Addr2> 
>                                        <Addr3 >STRTYPE</Addr3> 
>                                        <Addr4 >STRTYPE</Addr4> 
>                                        <Addr5 >STRTYPE</Addr5> 
>                                        <City >STRTYPE</City> 
>                                        <State >STRTYPE</State> 
>                                        <PostalCode >STRTYPE</PostalCode> 
>                                        <Country >STRTYPE</Country> 
>                                        <Note >STRTYPE</Note> 
>                                        <DefaultShipTo >BOOLTYPE</DefaultShipTo> 
>                                </ShipToAddress>```
>                                
> ```
Hello Greg, 
Sorry this is late, But I found one more property Missing here. the first property of ShipToAddress which is **Name**.

**Comment by sflagp** [comment]
Don't know why I missed that. Especially since it's a required property for a ShipToAddress. Published v1.2.10 to nuget. Should be there within an hour. 

[QbModels.1.2.10.zip](https://github.com/sflagp/QBXML-Object-Models/files/14103062/QbModels.1.2.10.zip)

**Comment by rashidabbas786** [comment]
I missed it too, Then I was looking for a key in the address and found this :-D
